pkg/graveler/ref/manager.go

```diff
@@ -629,3 +632,33 @@ func (m *Manager) DeleteExpiredLinkAddresses(ctx context.Context, repository *gr
 	}
 	return itr.Err()
 }
+
+func (m *Manager) DeleteExpiredImports(ctx context.Context, repository *graveler.RepositoryRecord) error {
+	expiry := time.Now().Add(-ImportExpiryTime)
+	repoPartition := graveler.RepoPartition(repository)
+	key := []byte(graveler.ImportsPath(""))
+	options := kv.IteratorOptionsFrom([]byte(""))
+	itr, err := kv.NewPrimaryIterator(ctx, m.kvStoreLimited, (&graveler.ImportStatusData{}).ProtoReflect().Type(), repoPartition, key, options)
+	if err != nil {
+		return fmt.Errorf("failed to get imports iterator from store: %w", err)
+	}
+	defer itr.Close()
+
+	for itr.Next() {
+		entry := itr.Entry()
+		status, ok := entry.Value.(*graveler.ImportStatusData)
+		if !ok {
+			return fmt.Errorf("invalid protobuf type %s: %w", entry.Value.ProtoReflect().Type().Descriptor().FullName(), graveler.ErrReadingFromStore)
+		}
+		if status.UpdatedAt.AsTime().Before(expiry) {
+			if !status.Completed && status.Error == "" {
+				logging.Default().WithFields(logging.Fields{"import_id": status.Id}).Warning("removing stale import")
```

Comment:
If any import we cancel will be later stale - I would go with Info or remove this one

Reply:
Canceled imports have a status.Error == "Canceled"